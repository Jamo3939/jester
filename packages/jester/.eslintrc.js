module.exports = {
  'parser': 'babel-eslint',
  'extends': [
    '@scotia/eslint-config-scotiabank'
  ],
  'env': {
    'jest': true
  }
};
