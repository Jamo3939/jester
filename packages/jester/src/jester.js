import forEach from 'lodash/forEach';

const runSnapshotTest = (name, object) => {
  test(`renders ${name} correctly`, () => {
    expect(JSON.stringify(object, null, 2)).toMatchSnapshot();
  });
};

const runSnapshotTests = (object) => forEach(
  object,
  (value, key) => runSnapshotTest(key, value)
);

export default {
  runSnapshotTests
};
