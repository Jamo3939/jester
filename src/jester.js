import React from 'react';
import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import forEach from 'lodash/forEach';

const getSimpleJsonTree = (Component, strategy) => {
  return toJSON(
    strategy(<Component>Snapshots Rock</Component>)
  );
};

const runSnapshotTest = (name, Component, strategy) => {
  test(`renders ${name} correctly`, () => {
    expect(getSimpleJsonTree(Component, strategy)).toMatchSnapshot();
  });
};

const buildSnapshotTestRunner = (strategy) => {
  return (components) => forEach(
    components,
    (component, key) => runSnapshotTest(key, component, strategy)
  );
};

export default {
  runDeepSnapshotTests: buildSnapshotTestRunner(mount),
  runShallowSnapshotTests: buildSnapshotTestRunner(shallow)
};
