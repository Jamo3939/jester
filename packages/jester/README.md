## jester

Jest helpers to reduce boilerplate.

## install

```js
npm install @scotia/jester
```

## usage

Provides methods that create and run tests for to compare json objects.

## api

#### #runSnapshotTests(Object : {})

Accepts any object and runs them through a simple snapshot test.

```js
import jester from '@scotia/jester';
describe('object snapshot', () => {
  jester.runSnapshotTests({
    one: { two: 'three' },
    four: { five: { six: 'seven' } }
  });
});
```

## development

#### Run tests

```bash
npm run test
npm run test:watch
```

#### Build

```bash
npm run build
```
