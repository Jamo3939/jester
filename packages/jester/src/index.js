import jester from './jester';
export default jester;
