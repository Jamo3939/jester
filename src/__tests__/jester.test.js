import { runSnapshotTests } from '../jester';

describe('jester', () => {
  describe('object snapshot', () => {
    runSnapshotTests({
      one: { two: 'three' },
      four: { five: { six: 'seven' } }
    });
  });
});
