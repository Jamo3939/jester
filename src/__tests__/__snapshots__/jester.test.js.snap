// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jester object snapshot renders four correctly 1`] = `
"{
  \\"five\\": {
    \\"six\\": \\"seven\\"
  }
}"
`;

exports[`jester object snapshot renders one correctly 1`] = `
"{
  \\"two\\": \\"three\\"
}"
`;
